samples/unit_testing/subscription-manager/unit_tests/test_createsubscription_suborchestrator.py

```diff
@@ -0,0 +1,176 @@
+import datetime
+import azure.durable_functions as df
+import azure.functions as func
+from datetime import timedelta
+from unittest import main
+from unittest import TestCase
+from unittest.mock import MagicMock, patch
+from ..Subscription.subscription import SubscriptionProperties, SubscriptionResponse
+from ..CreateSubscriptionSubOrchestrator import orchestrator_fn
+
+#######################################################################
+#### Call Activity Mocks ##############################################
+#######################################################################
+def call_activity_create_subscription(activityName: str, payload):
+    assert activityName == "CreateSubscription"
+    mock.call_activity.side_effect = call_activity_mock_status_check_succeeded
+
+def call_activity_mock_status_check_accepted(activityName: str, payload):
+    assert activityName == "StatusCheck"
+
+def call_activity_mock_status_check_succeeded(activityName: str, payload):
+    assert activityName == "StatusCheck"
+
+def call_activity_mock_status_check_notfound(activityName:str, payload):
+    mock.call_activity.side_effect = call_activity_create_subscription
+    assert activityName == "StatusCheck"
+
+def call_activity_mock_status_check_error(activityName:str, payload):
+    assert activityName == "StatusCheck"
+
+#######################################################################
+#### Create Timer Mocks ##############################################
+#######################################################################
+
+def create_timer_mock(next_checkpoint):
+    # check if next_checkpoint got scheduled 1 min more than previous checkpoint
+    assert int(next_checkpoint.strftime("%M")) == int(mock.current_utc_datetime.strftime("%M")) + 1
+
+    # change the call_activity side effect to succeeded call
+    mock.call_activity.side_effect = call_activity_mock_status_check_succeeded
```

Comment:
why is this necessary?

Reply:
I guess the confusion is because all the call_activity mocks are the same. While simulating a scenario where sub creation is already underway, where the subscription creation API's would respond with an "Accepted" state - the durable orchestrator will start a timer until this state changes from Accepted -> Succeeded. So I am mocking accordingly. What I can do to make this more meaningful is to return the canned response from each call activity to make them unique.

Reply:
Hmm, let's discuss this one during our meeting :)